Source/Core/DolphinQt2/RenderWidget.cpp

```diff
@@ -70,6 +73,14 @@ void RenderWidget::OnHideCursorChanged()
   setCursor(Settings::Instance().GetHideCursor() ? Qt::BlankCursor : Qt::ArrowCursor);
 }
 
+void RenderWidget::OnKeepOnTopChanged(bool top)
+{
+  setWindowFlags(top ? windowFlags() | Qt::WindowStaysOnTopHint :
+                       windowFlags() ^ Qt::WindowStaysOnTopHint);
```

Comment:
The second part should be `windowFlags() & ~Qt::WindowStaysOnTopHint` to not rely on the previous value.